{
  if (e instanceof ExecutionException && e.getCause() instanceof Exception)   return function.apply((Exception)e.getCause());
  return function.apply(e);
}
